tests/MongoDB.Bson.Tests/ObjectModel/Decimal128Tests.cs

```diff
@@ -121,6 +122,34 @@ public void ToDecimal_should_throw_when_value_is_out_of_range(string valueString
             exception.Should().BeOfType<OverflowException>();
         }
 
+        [Theory]
+        [InlineData("123.456", 123.456)]
+        public void ToDouble_should_not_depend_on_current_culture(string valueString, double expectedDouble)
+        {
+            var currentCulture = Thread.CurrentThread.CurrentCulture;
+            Thread.CurrentThread.CurrentCulture = new CultureInfo("fr-FR");
+            var subject = Decimal128.Parse(valueString);
+
+            var result = Decimal128.ToDouble(subject);
+
+            result.Should().Be(expectedDouble);
+            Thread.CurrentThread.CurrentCulture = currentCulture;
```

Comment:
Use a try/finally to ensure the CurrentCulture is reset even if an exception is thrown.

Same in next test.

Reply:
Done